### PR TITLE
Add liblzma (v5.6.2), fix which use

### PR DESCRIPTION
**Pull request by thesamesam**

### CODE REVIEW

**Comment by thesamesam** [comment]
See also https://github.com/compiler-explorer/infra/pull/1319. cc @dkm

**Review by mattgodbolt — APPROVED** [review]
Ideally these would be two PRs as they're unrelated. I'll merge though. thanks

**Comment by mattgodbolt** [comment]
I can't merge: the installer didn't work and I had to revert. Let me know if I can help. And, if you can, please separte this into a PR to fix the `which` and one to install the library.

**Comment by thesamesam** [comment]
I've split which into https://github.com/compiler-explorer/compiler-explorer/pull/6567.

File: etc/config/c.amazon.properties [hunk 1]
```diff
@@ -3161,6 +3161,15 @@ libs.libuv.versions.1381.version=1.38.1
 libs.libuv.versions.1381.path=/opt/compiler-explorer/libs/libuv/v1.38.1/include
 libs.libuv.versions.1381.libpath=/opt/compiler-explorer/libs/libuv/v1.38.1/x86_64/lib:/opt/compiler-explorer/libs/libuv/v1.38.1/x86/lib
 
+libs.liblzma.name=liblzma
+libs.liblzma.description=XZ-format compression library
+libs.liblzma.liblink=lzma
+libs.liblzma.url=https://tukaani.org/xz/
+libs.liblzma.versions=562
+libs.liblzma.versions.562.version=5.6.2
+libs.liblzma.versions.562.path=/opt/compiler-explorer/libs/liblzma/v5.6.2/include
```
Comment (mattgodbolt) [inline etc/config/c.amazon.properties:3170]: I'm going to merge and patch up based on how I think this needs to go

**Review by mattgodbolt — APPROVED** [review]

File: etc/config/c.amazon.properties (hunk 1 above)
Comment (thesamesam) [reply etc/config/c.amazon.properties:3170]: Thank you Matt for the handholding - I didn't expect it to be so much hassle!

**Comment by mattgodbolt** [comment]
So... I've been trying to get this to work and I have for now failed :(  I filed https://github.com/compiler-explorer/compiler-explorer/issues/6585 to support C and maybe I'll be able to pick it up later.

I can probably get it working for C++ if that's useful?

**Pull request merged**